### ISSUE TITLE
Base URL not dynamic

### ISSUE THREAD

**Issue by erard22** [eclipse/microprofile-rest-client#213]
With the current specification the `baseURL` is static and cannot be changed during runtime. For simple scenarios or systems running on kubernetes this might be fine. But imagine a system using Consul for service discovery, there we might change the base url on the fly during runtime depending on the health of the endpoints. And depending on the underlying implementation changing the target should be possible without performance impact. Even the client is `ApplicationScoped`. The client should set the target with every request or check if the `baseURL` has changed. The change in mp-config could be done by an extension and must not be seen as static.

**Comment by kenfinnigan** [comment]
Do you mean you want this functionality with injecting a CDI Bean of the interface, as it's already possible to do this with the programmatic API?

**Comment by erard22** [comment]
Basically I would like to have an `ApplicationScoped` client that reacts on changes in the mp-config (at least for the URL), what is possible via the different mp-config extensions. Programmatically I can create a client with every request and use the current configured url then the client is `RequestScoped`
But digging more in the code this is probably not feasible. At least with RestEasy a new `WebTarget` and consequently a new proxy is needed.

**Comment by kenfinnigan** [comment]
Correct. RESTEasy builder sets the URL fairly early on which wouldn’t allow for reuse of the instance

**Comment by sopgreg** [comment]
+1

The URL (including protocol and port) should be easily changeable at runtime for the reasons already mentioned. We also do health monitoring on endpoints and need to switch URLs at runtime accordingly.

**Comment by OleKsimov** [comment]
Would be great to have this feature implemented. Right now using microprofile rest client for requests with different URLs is not quite possible. I receive a callback URL inside each request, create rest client every time for each callback url is not an option

**Comment by hbzhou** [comment]
I have the same use case.  One more thing,  HTTP clients are heavy objects, and creating one on each request causes overhead
